extensions/amp-story/1.0/amp-story.js

```diff
@@ -1089,28 +1097,39 @@ export class AmpStory extends AMP.BaseElement {
   }
 
   /**
-   * Displays the unsupported browser UI: either the publisher provided UI, or
-   * fallbacks to a generic default.
+   * If browser is supported, displays the story. Otherwise, shows either the
+   * default unsupported browser layer or the publisher fallback (if provided).
    * @param {boolean} isBrowserSupported
    * @private
    */
   onSupportedBrowserStateUpdate_(isBrowserSupported) {
-    if (isBrowserSupported) {
-      dev().error(TAG, 'No handler to exit unsupported browser state.');
-    }
-
     const fallbackEl = this.getFallback();
-
-    this.mutateElement(() => {
-      this.element.classList.add('i-amphtml-story-fallback');
-    });
-
-    // Displays the publisher provided fallback, or fallbacks to the default
-    // unsupported browser layer.
-    if (fallbackEl) {
-      this.toggleFallback(true);
+    if (isBrowserSupported) {
+      // Removes the default unsupported browser layer or throws an error
+      // if the publisher has provided their own fallback
+      if (fallbackEl) {
+        dev().error(TAG, 'No handler to exit unsupported browser state on ' +
+        'publisher provided fallback.');
+      } else {
+        //removeElement(this.unsupportedBrowserLayer_.get());
+        this.unsupportedBrowserLayer_.remove();
+        this.mutateElement(() => {
+          this.element.classList.remove('i-amphtml-story-fallback');
+        });
+        this.layoutStory_();
+      }
     } else {
-      this.element.appendChild(this.unsupportedBrowserLayer_.build());
+      this.mutateElement(() => {
+        this.element.classList.add('i-amphtml-story-fallback');
+      });
+
+      // Displays the publisher provided fallback, or fallbacks to the default
+      // unsupported browser layer.
+      if (fallbackEl) {
+        this.toggleFallback(true);
+      } else {
+        this.element.appendChild(this.unsupportedBrowserLayer_.build());
```

Comment:
I think this should wrapped be in a `this.mutateElement(...)` call.

Reply:
A separate `this.mutateElement` with just this line should be good!

And yes, I think there are probably a lot of mutates that we've missed along the way 😦.  Would be good to audit them all, but in the meantime it's also good if we can fix any that we see along the way!

Reply:
I did end up wrapping this line in the mutateElement, but I did end up adding the .get() method we talked about in the unsupported-browser-layer.js file. I couldn't get the tests to pass without building the layer outside the mutateElement, and then only appending the child inside the mutateElement() call. 

You can see this in my newest changes, and I understand that that might not be good of a justification to make the get() method haha.
 
So please let me know if there is something you see that I'm missing that would make simply wrapping that line in the mutateElement() call work!